### DISCUSSION TITLE
scipy and numpy need to apt install on lite

### DISCUSSION THREAD

**Discussion by paddywwoof** [piwheels/piwheels#337] [General]
Hi the front page of piwheels.org gives an example of installing scipy but doesn't mention the need for `sudo apt install libgfortran5 libopenblas0-pthread` on bookworm lite this will error when `import scipy`

I know that if installing on the desktop version of Raspberry Pi those two libraries will already be there... but so will lots of other python modules such as numpy and pil.

**Comment by bennuttall** [comment]
We include apt dependencies on the project pages:

- https://piwheels.org/project/numpy/
- https://piwheels.org/project/scipy/

We should probably change the example on the homepage to something that would normally require compilation but doesn't have any apt dependencies.

**Reply by paddywwoof** [reply]
Ben, yes that's where I found the solution to the missing blas library error messages*. I just noticed afterwards that the front page had an example that I suspected wouldn't work on lite.

I don't remember this issue in the past. pi3d just has numpy and pillow as dependencies and I thought that they 'just worked' when the user `python -m pip install pi3d` on bullseye lite, say.

*the python error message has a link to the numpy website https://numpy.org/doc/stable/user/troubleshooting-importerror.html#raspberry-pi where one suggestion (after apt installing libatlas-base-dev - would this be a simpler alternative in your instructions?) is
```
pip3 uninstall numpy  # remove previously installed version
apt install python3-numpy
```
!!

**Reply by bennuttall** [reply]
This is the process of determining dependencies: https://blog.piwheels.org/how-to-work-out-the-missing-dependencies-for-a-python-package/

We use an automated version of this approach to calculate the dependencies.

**Reply by paddywwoof** [reply]
Thanks, that's very clear and helpful. It might be an idea to update those instructions for /home/myuserid/myvenv/lib/python3.11/site-packages/<module> now venv is compulsory, and having a little script to go through all the subdirectories would be essential for modules such as numpy which has dispersed all the .so files in a tree structure.
